langwatch/src/components/traces/TraceDetails.tsx

```diff
@@ -80,6 +90,35 @@ export function TraceDetails(props: {
     }
   );
 
+  const annotationQueues = api.annotation.getQueues.useQuery(
+    { projectId: project?.id ?? "" },
+    {
+      enabled: !!project,
+    }
+  );
+
+  const users =
+    api.organization.getOrganizationWithMembersAndTheirTeams.useQuery(
+      {
+        organizationId: organization?.id ?? "",
+      },
+      {
+        enabled: !!organization,
+      }
+    );
+
+  const userOptions = users.data?.members.map((member) => ({
+    label: member.user.name ?? "",
+    value: `user-${member.user.id}`,
+  }));
+
+  const queueOptions = annotationQueues.data?.map((queue) => ({
+    label: queue.name ?? "",
+    value: `queue-${queue.id}`,
+  }));
+
+  const options = [...(userOptions ?? []), ...(queueOptions ?? [])];
```

Comment:
trace details is already very busy, I think it's better to extract the logic for adding to queues to it's own component, also because later we want to use it on the table selection batch send to queue (like when you select many traces to add to dataset). Can you extract this data fetching, the popover, the AddParticipants and everything else to a separate component and just use it here? (can be done in a follow up PR too)